browser/src/core/Socket.js

```diff
@@ -594,17 +594,17 @@ app.definitions.Socket = L.Class.extend({
 
 		var isTile = e.textMsg.startsWith('tile:');
 		var isDelta = e.textMsg.startsWith('delta:');
-		if (!isTile && !isDelta &&
-		    !e.textMsg.startsWith('renderfont:') &&
-			!e.textMsg.startsWith('slidelayer:') &&
-		    !e.textMsg.startsWith('windowpaint:'))
+		var isSlideLayer = e.textMsg.startsWith('slidelayer:');
+		if (!isTile && !isDelta && !isSlideLayer && !e.textMsg.startsWith('renderfont:') &&
+			!e.textMsg.startsWith('windowpaint:'))
 			return;
 
 		if (e.textMsg.indexOf(' nopng') !== -1)
 			return;
 
 		// pass deltas through quickly.
-		if (e.imgBytes && (isTile || isDelta) && e.imgBytes[e.imgIndex] != 80 /* P(ng) */)
+		if (e.imgBytes && (isTile || isDelta || isSlideLayer) &&
+			e.imgBytes[e.imgIndex] != 80 /* P(ng) */)
```

Comment:
I think this "e.imgBytes[e.imgIndex] != 80" line only makes sense for isTile or isDelta and not for isSlideLayer and I guess there's a danger that by chance the first byte of the raw slidelayer compressed stream might be 80 and cause it to be misdetected.